### ISSUE TITLE
UE4 crashes at beginning stating an issue in StartupModule function of the plugin

### ISSUE THREAD

**Issue by parthnaik92** [N7Alpha/UnrealLibretro#11]
Hey @N7Alpha, 
I tried incorporating the plugin in a blank C++ project after you updated the plugin to support Android. But it is showing the following error screen while opening the project. I am using UE 4.26 and VS 2019. The error is on line 37 in [UnrealLibretro.cpp](https://github.com/N7Alpha/UnrealLibretro/blob/master/Source/UnrealLibretro/Private/UnrealLibretro.cpp#:~:text=int%20load_sdl_error%20%3D%20SDL_Init(SDL_INIT_VIDEO)%3B) in the StartupModule function when it tries to initialize SDL.

![image](https://user-images.githubusercontent.com/4035348/152900975-ebd21c7e-6ab4-4765-82dc-dcab2cd456d6.png)

I followed your guide from the readme.
1. Downloaded the latest source
2. Used the "using git" method to install it to the project. The setup.sh file ran without any issues and it downloaded all the other dependencies.
3. The project compiled successfully from Visual Studio.
4. The project crashed at around 75% with the error message shown in the image above.

Can you please help with this issue?
Thanks!

**Comment by N7Alpha** [comment]
My bad I removed something I probably shouldn't have although it still _worked on my machine_. I've pushed a commit that hopefully fixes it. You should just be able to pull it to try it out.

If you try out the Android stuff let me know how that goes.

**Comment by parthnaik92** [comment]
Thanks a lot! I was able to open the project now without any issues by using the updated plugin. I will keep you updated about Android stuff, will most probably tinker with it during post-work hours or the weekend.

Keeping the issue open till then.

**Comment by parthnaik92** [comment]
I was able to make it work on my Quest. Thanks again!
I will let you know if I run into any other issues. Currently, while adding inputs to the key binding maps, I am unable to access any Oculus Touch related keys (except 4 captouch ones). But I think that issue is of UE4, not related to this plugin.

![Libretro Quest Test](https://user-images.githubusercontent.com/4035348/153064798-c6ca8bc7-b961-438b-966b-5281290e80c2.jpg)

**Comment by AltoRetrato** [comment]
Hi, @parthnaik92 and @N7Alpha !

I'm having issues running UnrealLibreto on the Quest 2. Could you tell me how you got it to work?

I can build, deploy and run UE projects to the Oculus / Meta Quest 2 VR headset, but I'm not sure how to compile the UnrealLibreto for it. I can compile it for Windows, though, with:
```
RunUAT.bat BuildPlugin -plugin="[path to plugin.uplugin]" -package="[output path]" -TargetPlatforms=Win64 -VS2019
```

I tried building it for the Quest 2 / Android with:
```
RunUAT.bat BuildPlugin -plugin="[path to plugin.uplugin]" -package="[output path]" -TargetPlatforms=Android -rocket
```
(... but I could not see any difference running it with or without the `-rocket` parameter.)

In the end, even though I set `-TargetPlatforms=Android` it built the plugin for Windows (`Building plugin for host platforms: Win64`) and seemed to build it for Android (`Building plugin for target platforms: Android`), but during the build process it showed `Target: x86_64-w64-windows-gnu` several times while building for Android. Closer to the end, it looked more promising:
```
Building 2 actions with 2 processes...
[1/2] Compile pixconv.c [arm64]
[2/2] Compile Module.UnrealLibretro.cpp [arm64]
```
... and ended with:
```
4 warnings generated.
Total time in Parallel executor: 10.55 seconds
Total execution time: 11.87 seconds
Took 11.924294s to run dotnet.exe, ExitCode=0
Reading filter rules from E:\Unreal\LibretroVR_511\Plugins\Output\HostProject\Plugins\UnrealLibretro\Config\FilterPlugin.ini
Loading FilterPluginAndroid.ini
BUILD SUCCESSFUL
AutomationTool executed for 0h 2m 49s
AutomationTool exiting with ExitCode=0 (Success)
```
But in the `Output\Binaries` folder there is only the `Win64` folder. In the `Output\Intermediate\Build` there are `Win64` and `Android` folders. I found the Android ELF files in `Output\Intermediate\Build\Android\UnrealGame\Development\UnrealLibretro` and `Output\Intermediate\Build\Android\UnrealGame\Shipping\UnrealLibretro`, e.g.:
```
5,254,800 Module.UnrealLibretro.cppa8.o
   58,720 pixconv.ca8.o
       82 UnrealLibretro.precompiled
```
When I compile, deploy and run the APK in the headset, no matter if I copy the ELF files to `Plugins\Binaries\Android` of `Plugins\Intermediate\Build\Android\...`, it crashes. Here's the output log:
```
Error: UAT: 03-30 19:14:00.339 25466 25563 D UE      : LogPluginManager: Error: Unable to load plugin 'UnrealLibretro'. Aborting.
UAT: 03-30 19:14:00.375 25466 25563 D UE      : Assertion failed: false [File:D:/build/++UE5/Sync/Engine/Source/Runtime/Launch/Private/Android/LaunchAndroid.cpp] [Line: 529]
UAT: 03-30 19:14:00.376 25466 25563 D UE      : Engine Preinit Failed
UAT: 03-30 19:14:00.379 25466 25563 D UE      : Assertion failed: false [File:D:/build/++UE5/Sync/Engine/Source/Runtime/Launch/Private/Android/LaunchAndroid.cpp] [Line: 529]
UAT: 03-30 19:14:00.380 25466 25563 D UE      : Engine Preinit Failed
Error: UAT: 03-30 19:14:00.380 25466 25563 D UE      : LogAndroid: Error: === Critical error: ===
Error: UAT: 03-30 19:14:00.380 25466 25563 D UE      : LogAndroid: Error:
Error: UAT: 03-30 19:14:00.380 25466 25563 D UE      : LogAndroid: Error: Assertion failed: false [File:D:/build/++UE5/Sync/Engine/Source/Runtime/Launch/Private/Android/LaunchAndroid.cpp] [Line: 529]
Error: UAT: 03-30 19:14:00.380 25466 25563 D UE      : LogAndroid: Error: Engine Preinit Failed
Error: UAT: 03-30 19:14:00.380 25466 25563 D UE      : LogAndroid: Error: [Callstack] 0x0000007B59C95B38 libUnreal.so(0x0000000009F48B38)!AndroidMain(android_app*)  []
Error: UAT: 03-30 19:14:00.380 25466 25563 D UE      : LogAndroid: Error: [Callstack] 0x0000007B59CA4448 libUnreal.so(0x0000000009F57448)!android_main()  []
Error: UAT: 03-30 19:14:00.380 25466 25563 D UE      : LogAndroid: Error: [Callstack] 0x0000007B59CDB380 libUnreal.so(0x0000000009F8E380)![Unknown]()  []
Error: UAT: 03-30 19:14:00.380 25466 25563 D UE      : LogAndroid: Error: [Callstack] 0x0000007C522D5C48 libc.so(0x00000000000D5C48)![Unknown]()  []
Error: UAT: 03-30 19:14:00.380 25466 25563 D UE      : LogAndroid: Error: [Callstack] 0x0000007C52282FE8 libc.so(0x0000000000082FE8)![Unknown]()  []
Error: UAT: 03-30 19:14:00.380 25466 25563 D UE      : LogAndroid: Error:
```

Any help would be appreciated. Thanks!

**Comment by AltoRetrato** [comment]
Thanks!

After banging my head on the wall a few days, I made a breakthrough! :)

Instead of adding the plugin source to a Blueprint project (the VR template is BP only, it seems), I created a blank C++ project to build the plugin and move the plugin folder to the BP project.

I still had a few issues / crashes when the plugin couldn't load the ROM. One of the reasons was because I was using a backslash ("folder\rom") instead of a forward slash ("folder/rom") in the ROM path (which works on Windows, but not on Android).

I hope this helps anybody else with issues trying to do the same.
